chains/evm/contracts/test/onRamp/CCVProxy/CCVProxy.parseExtraArgsWithDefaults.t.sol

```diff
@@ -0,0 +1,224 @@
+      tokenArgs: ""
+    });
+
+    bytes memory encodedExtraArgs = abi.encodePacked(Client.GENERIC_EXTRA_ARGS_V3_TAG, abi.encode(extraArgs));
+
+    Client.EVMExtraArgsV3 memory result =
+      s_ccvProxyTestHelper.parseExtraArgsWithDefaults(destChainConfig, encodedExtraArgs);
+
+    assertEq(result.requiredCCV.length, 2); // 1 from config + 1 default
+    assertEq(result.requiredCCV[0].ccvAddress, REQUIRED_CCV); // Required CCV from config
+    assertEq(result.requiredCCV[1].ccvAddress, DEFAULT_CCV); // Default CCV
+    assertEq(result.executor, DEFAULT_EXECUTOR); // Default executor
+  }
+
+  function test_parseExtraArgsWithDefaults_WithLegacyExtraArgs() public {
+    CCVProxy.DestChainConfig memory destChainConfig = s_defaultDestChainConfig;
+
+    bytes memory legacyExtraArgs = "legacy args";
+
+    Client.EVMExtraArgsV3 memory result =
+      s_ccvProxyTestHelper.parseExtraArgsWithDefaults(destChainConfig, legacyExtraArgs);
+
+    assertEq(result.requiredCCV.length, 2); // 1 from config + 1 default
+    assertEq(result.requiredCCV[0].ccvAddress, REQUIRED_CCV); // Required CCV from config
+    assertEq(result.requiredCCV[1].ccvAddress, DEFAULT_CCV); // Default CCV with legacy args
+    assertEq(result.requiredCCV[1].args, legacyExtraArgs);
+    assertEq(result.executorArgs, legacyExtraArgs);
+    assertEq(result.executor, DEFAULT_EXECUTOR); // Default executor
+  }
+
+  function test_parseExtraArgsWithDefaults_RevertWhen_InvalidOptionalCCVThreshold() public {
+    CCVProxy.DestChainConfig memory destChainConfig = CCVProxy.DestChainConfig({
+      router: s_sourceRouter,
+      sequenceNumber: 1,
+      requiredCCV: address(0),
+      defaultCCV: DEFAULT_CCV,
+      defaultExecutor: DEFAULT_EXECUTOR
+    });
+
+    Client.CCV[] memory optionalCCVs = new Client.CCV[](2);
+    optionalCCVs[0] = Client.CCV({ccvAddress: s_optionalCCV1, args: "optional1"});
+    optionalCCVs[1] = Client.CCV({ccvAddress: s_optionalCCV2, args: "optional2"});
+
+    Client.EVMExtraArgsV3 memory extraArgs = Client.EVMExtraArgsV3({
+      requiredCCV: new Client.CCV[](0),
+      optionalCCV: optionalCCVs,
+      optionalThreshold: 2, // Invalid: threshold >= optionalCCV.length
+      finalityConfig: 10,
+      executor: address(0),
+      executorArgs: "",
+      tokenArgs: ""
+    });
+
+    bytes memory encodedExtraArgs = abi.encodePacked(Client.GENERIC_EXTRA_ARGS_V3_TAG, abi.encode(extraArgs));
+
+    vm.expectRevert(CCVProxy.InvalidOptionalCCVThreshold.selector);
+    s_ccvProxyTestHelper.parseExtraArgsWithDefaults(destChainConfig, encodedExtraArgs);
+  }
+
+  function test_parseExtraArgsWithDefaults_RevertWhen_ZeroOptionalThreshold() public {
+    CCVProxy.DestChainConfig memory destChainConfig = CCVProxy.DestChainConfig({
+      router: s_sourceRouter,
+      sequenceNumber: 1,
+      requiredCCV: address(0),
+      defaultCCV: DEFAULT_CCV,
+      defaultExecutor: DEFAULT_EXECUTOR
+    });
+
+    Client.CCV[] memory optionalCCVs = new Client.CCV[](2);
```

Comment:
Please always try to write minimal tests. This would already fail with a single optional CCV so you can remove the second one.